File: src/01-easy/the-river-ii-/solution.java
import java.util.*;
class Solution {
    static int ds(long n){int s=0;while(n>0){s+=(int)(n%10);n/=10;}return s;}
    public static void main(String[] a){
        long r=new Scanner(System.in).nextLong();
        for(long x=Math.max(1,r-45);x<r;x++)if(x+ds(x)==r){System.out.println("YES");return;}
        System.out.println("NO");
    }
}

File: src/01-easy/the-river-ii-/solution.clj
(ns Solution (:gen-class))
(defn ds [n] (reduce + (map #(- (int %) 48) (str n))))
(defn -main [& args]
  (let [r (read)
        lo (max 1 (- r 45))]
    (println (if (some #(= (+ % (ds %)) r) (range lo r)) "YES" "NO"))))
